src/components/Selector/LanguageSelector.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useRef, useState } from 'react';
+import { Button } from '../Button';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import Divider from '@mui/material/Divider';
+
+interface DropdownProps {
+  options: string[];
+  onSelect: (value: string) => void;
+}
+
+function LanguageSelector({ options, onSelect }: DropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isSearching, setIsSearching] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [filtedLanguages, setFiltedLanguages] = useState<string[]>([]);
+  const [languageQuery, setLanguageQuery] = useState('');
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const toggleDropdown = () => setIsOpen(!isOpen);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+        setFiltedLanguages([]);
+        setIsSearching(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [dropdownRef]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setLanguageQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    if (languageQuery == '') {
+      setIsSearching(false);
+      setFiltedLanguages([]);
+    }
+    if (languageQuery.length > 0) {
+      setIsSearching(true);
+      setFiltedLanguages(
+        options.filter((language) =>
+          language.toLowerCase().includes(languageQuery.toLowerCase())
+        )
+      );
+    }
+  }, [languageQuery]);
```

Comment:
I dont think a hook is necessary here. Is it can be a normal function called when handling input?